src/logplex_tlssyslog_drain.erl

```diff
@@ -404,6 +405,23 @@ do_reconnect(State = #state{sock = undefined,
                     %% first failure.
                     ok;
                 _ ->
+                    % inject log message
+                    Msg =  iolist_to_binary(
+                               logplex_syslog_utils:to_msg(
+                                 logplex_syslog_utils:fmt(local7,
+                                                          warning,
+                                                          now,
+                                                          "heroku", 
+                                                          "logplex",
+                                                          "Error L14 (certificate validation): "
+                                                          "error=\"~s" uri=\"~s\"",
+                                                          [error_to_human(Reason), drain_uri(State)]),
+                                 State#state.drain_tok)),
+                    logplex_message:process_msg(Msg,
+                                                State#state.channel_id,
+                                                <<"heroku">>,
+                                                <<"heroku">>,
+                                                logplex_message:shard_info()),
```

Comment:
OK, as I look at this more, I tend to think that this is out-of-place code. We're in a drain doing message output to syslog+tls, but now wanting to re-inject a message into a channel. Message input and message output are pretty well separated in Logplex, and I think it'd be good to keep it that way. Is there already an internal 'I need to inject a message into a channel' function? If not, there probably should be (and would be more or less this code right?).